### PR TITLE
Block auto-disarm when in Launch Detection for Catapault/Hand-Launched FW

### PR DESCRIPTION
**Pull request by Kjkinney**
**Describe problem solved by the proposed pull request**
Hand launched Fixed-Wings are unable to use auto-disarm due to it being based of the Land-Detector. Because the Launch-Detection is less sensitive to "Non-Landed" States than the Land-Detector, this will often lead to a situation where the motor is not spinning and the vehicle transitions from landed to non-landed and then back to landed, triggering the auto disarm. This occurs during ground handling after the vehicle is armed. Currently we leave auto-disarm disabled.

**Test data / coverage**
This has been tested in a lab on a pixracer flight controller mounted on a fixed wing vehicle while spoofing GPS. https://review.px4.io/plot_app?log=e6249163-72e1-4ae6-9aa0-b56c947659de

**Describe your preferred solution**
This issue is solved by having the Fixedwing Position Controller set a message on the position_controller_status uORB struct determining whether or not Launch-Detection is running. This message is then subscribed to by the commander, which then blocks all auto-disarming if it is running(true). 

**Describe possible alternatives**
There was talk about using a variable already declared for this, but that is based off of Land-Detector which this is circumventing. 

**Additional context**
Discussed in this weeks dev-call.

FYI @Antiheavy

### CODE REVIEW

**Comment by Antiheavy** [comment]
FYI/A @MaEtUgR , we discussed this during the Dev call.  This PR helps make life easier for hand-launched fixed wings, now that auto-disarm is enabled by default: https://github.com/PX4/Firmware/pull/12346

File: src/modules/commander/Commander.cpp [hunk 1]
```diff
@@ -1317,6 +1322,10 @@ Commander::run()
 	int32_t airmode = 0;
 	int32_t rc_map_arm_switch = 0;
 
+	/* Catapault Launch Enabled */
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp]:
```suggestion
	/* Catapult Launch Enabled */
```

File: src/modules/commander/Commander.cpp [hunk 2]
```diff
@@ -3832,6 +3843,30 @@ void Commander::data_link_check(bool &status_changed)
 	}
 }
 
+
+bool Commander::launch_detection_running()
+{
+	bool updated;
+
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp]:
```suggestion
```

File: src/modules/commander/Commander.cpp [hunk 3]
```diff
@@ -3832,6 +3843,30 @@ void Commander::data_link_check(bool &status_changed)
...
+	position_controller_status_s _controller_status{};
+
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp]:
```suggestion
```

File: src/modules/commander/Commander.cpp [hunk 4]
```diff
@@ -3832,6 +3843,30 @@ void Commander::data_link_check(bool &status_changed)
 	}
 }
 
+
+bool Commander::launch_detection_running()
+{
+	bool updated;
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp]:
```suggestion
	bool updated = false;
```

File: src/modules/commander/Commander.cpp [hunk 5]
```diff
@@ -3832,6 +3843,30 @@ void Commander::data_link_check(bool &status_changed)
...
+
+	position_controller_status_s _controller_status{};
+
+	orb_check(_position_controller_sub, &updated);
+
+	if (updated) {
+
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp]:
```suggestion
```

File: src/modules/commander/Commander.cpp [hunk 6]
```diff
@@ -3832,6 +3843,30 @@ void Commander::data_link_check(bool &status_changed)
...
+		orb_copy(ORB_ID(position_controller_status), _position_controller_sub, &_controller_status);
+
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp]:
```suggestion
```

File: src/modules/commander/Commander.cpp [hunk 7]
```diff
@@ -3832,6 +3843,30 @@ void Commander::data_link_check(bool &status_changed)
...
+
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp]:
```suggestion
```

File: src/modules/commander/Commander.cpp [hunk 8]
```diff
@@ -3832,6 +3843,30 @@ void Commander::data_link_check(bool &status_changed)
...
+	}
+
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp]:
```suggestion
```

File: src/modules/commander/Commander.cpp [hunk 9]
```diff
@@ -3832,6 +3843,30 @@ void Commander::data_link_check(bool &status_changed)
...
+
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp]:
```suggestion
```

File: src/modules/commander/Commander.cpp [hunk 10]
```diff
@@ -3832,6 +3843,30 @@ void Commander::data_link_check(bool &status_changed)
...
+
+	return _controller_status.launch_detection_running;
+
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp]:
```suggestion
```

File: src/modules/commander/Commander.cpp [hunk 11]
```diff
@@ -3832,6 +3843,30 @@ void Commander::data_link_check(bool &status_changed)
...
+}
+
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp]:
```suggestion
```

File: src/modules/commander/Commander.cpp [hunk 12]
```diff
@@ -3832,6 +3843,30 @@ void Commander::data_link_check(bool &status_changed)
...
+
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp]:
```suggestion
```

File: src/modules/commander/Commander.hpp [hunk 13]
```diff
@@ -196,6 +197,9 @@ class Commander : public ModuleBase<Commander>, public ModuleParams
 
 	void update_control_mode();
 
+
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.hpp]:
```suggestion
```

File: src/modules/commander/Commander.cpp [hunk 14]
```diff
@@ -1251,6 +1254,8 @@ Commander::run()
 	uORB::Subscription system_power_sub{ORB_ID(system_power)};
 	uORB::Subscription vtol_vehicle_status_sub{ORB_ID(vtol_vehicle_status)};
 
+
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp]:
```suggestion
```

File: src/modules/commander/Commander.cpp [hunk 15]
```diff
@@ -1251,6 +1254,8 @@ Commander::run()
...
+
```
Comment (MaEtUgR) [inline src/modules/commander/Commander.cpp]:
```suggestion
```

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp [hunk 16]
```diff
@@ -2036,3 +2042,26 @@ int fw_pos_control_l1_main(int argc, char *argv[])
 {
 	return FixedwingPositionControl::main(argc, argv);
 }
+
+bool FixedwingPositionControl::_get_launchdetection_status(const position_setpoint_s &pos_sp_curr)
+{
+
+ bool launch_detection_running;
+
+ if(pos_sp_curr.type == position_setpoint_s::SETPOINT_TYPE_TAKEOFF &&
```
Comment (MaEtUgR) [inline src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp]: style incorrect in this method

**Comment by Kjkinney** [comment]
Re-based and implemented the suggested improvements. @MaEtUgR can you review it again?

File: src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp [hunk 17]
```diff
@@ -1721,6 +1723,10 @@ FixedwingPositionControl::run()
 			parameters_update();
 		}
 
+		//Place the function here
```
Comment (Antiheavy) [inline src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp]: @Kjkinney   is this needed?

Comment (Kjkinney) [reply src/modules/fw_pos_control_l1/FixedwingPositionControl.cpp]: Nope! Removed

**Comment by Antiheavy** [comment]
We have tested this branch on the ground.  We also have been extensively flight testing a slightly different version of this (built on v1.8.0) for the past month.  It works quite well.

**Comment by RomanBapst** [comment]
@Kjkinney Could you please resolve the conflict so that we can merge it?

**Comment by Kjkinney** [comment]
Moving to new PR

**Pull request closed without merging**